cmd/makedocs/main.go

```diff
@@ -2,48 +2,73 @@ package main
 
 import (
 	"bytes"
+	"fmt"
+	"go/parser"
+	"go/token"
+	"html/template"
 	"io/ioutil"
 	"log"
+	"os"
 	"strings"
-	"text/template"
 
 	"github.com/go-critic/go-critic/lint"
 )
 
 const (
 	docsPath      = "../../docs/"
-	checkersPath  = docsPath + "checkers/"
 	templatesPath = docsPath + "templates/"
+	checkersPath  = "../../lint/"
 )
 
 type checker struct {
 	Name             string
 	ShortDescription string
 	Description      string
+	Before           string
+	After            string
 	SyntaxOnly       bool
 	Experimental     bool
 	VeryOpinionated  bool
 }
 
+var checkers []checker
+
 func main() {
 	tmpl, err := template.ParseFiles(templatesPath + "overview.md.tmpl")
 	if err != nil {
 		log.Fatal(err)
 	}
-	checkers := []checker{}
+	pkgs, err := parser.ParseDir(&token.FileSet{}, checkersPath,
+		func(inf os.FileInfo) bool {
+			return strings.HasSuffix(inf.Name(), "_checker.go")
+		}, parser.ParseComments)
+	if err != nil {
+		log.Fatal(err)
+	}
+
 	for _, r := range lint.RuleList() {
-		desc, err := getDesc(r.Name())
-		if err != nil {
-			log.Fatal(r.Name())
+		log.Printf("parsing %s\n", r.Name())
+
+		f, ok := pkgs["lint"].Files[fmt.Sprintf("%s%s_checker.go", checkersPath, r.Name())]
+		if !ok {
+
+			log.Printf("File not found: %s%s_checker.go", checkersPath, r.Name())
+			continue
+		}
+		c := checker{
+			Name:            r.Name(),
+			SyntaxOnly:      r.SyntaxOnly,
+			Experimental:    r.Experimental,
+			VeryOpinionated: r.VeryOpinionated,
+		}
+		for _, comment := range f.Comments {
+			if strings.HasPrefix(comment.Text(), "!") {
```

Comment:
`Text()` computes combined text every time you call it.
Maybe it's enough to check first comment from group test for that prefix?
Or re-use `Text()` result.
Just making sure you know what's going on here. There should be no real performance issue here.

Reply:
From the other side: how many comments it will process? Hundreds at most ;)

Reply:
Yep, this is why it doesn't matter.
Just making a note that `Text()` is not a simple getter.